src/App.js

```diff
@@ -1,26 +1,700 @@
+            id="agencyTests"
+            onChange={this.testsSearchHandler}
+          >
+            <option value="Void">...</option>
+            <option value="agency_retrieve">Agency Retrieve</option>
+            <option value="agency_bind">Agency Bind</option>
+            <option value="agency_underwriting">Agency Underwriting</option>
+          </select>
+        </div>
+      ) : this.state.showApp === 'csr' ? (
+        <div>
+          <p>Test name</p>
+          <select
+            value={this.state.showTests}
+            id="csrTests"
+            onChange={this.testsSearchHandler}
+          >
+            <option value="Void">...</option>
+            <option value="csrPdfVerify">CSR PDF Verify</option>
+            <option value="csrBind">CSR Bind</option>
+            <option value="csrPolicy">CSR Policy</option>
+            <option value="csrEndorsements">CSR Endorsements</option>
+          </select>
+        </div>
+      ) : null;
+
+    let envSelect =
+      this.state.showTests === 'Void' ? null : (
+        <div>
+          {' '}
+          <p> Environment name</p>
+          <select
+            value={this.state.showEnv}
+            id="env"
+            onChange={this.envSearchHandler}
+          >
+            <option value="Void">...</option>
+            <option value="staging">Staging</option>
+            <option value="demo">Demo</option>
+            <option value="prod">Production</option>
+          </select>
+        </div>
+      );
+
+    let table = [];
+
+    let nav = [];
+    let sideBar = null;
+
+    if (this.state.showTable !== false) {
+      for (let i = 0; i < content.length; i++) {
+        table.push(
+          <Table
+            content={content[i]}
+            number={this.state.testResults[i].release}
+          />
+        );
+      }
+
+      for (let i = 1; i <= 14; i++) {
+        nav.push(i);
+      }
+
+      sideBar = <Navigation count={nav} />;
+    }
+
+    return (
+      <div className="App">
+        <header>
+          <h1>Test Results Page</h1>
+          <h3> Please select the application, the test and the environment </h3>
+
+          <p>Application name</p>
+          <select id="application" onChange={this.searchHandler}>
+            <option value="Void">...</option>
+            <option value="agency">Agency</option>
+            <option value="consumer">Consumer</option>
+            <option value="csr">CSR</option>
+          </select>
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+          {testSelect}
```

Comment:
Each of these can be components instead of variables. There is technically nothing wrong with the way you have done this, but it will become a HUGE pain to maintain, trust me :)